### ISSUE TITLE
False positive in SwitchDeclaration TerminatingComment with all branches covered

### ISSUE THREAD

**Issue by simPod** [squizlabs/PHP_CodeSniffer#2800]
```php
switch ($x) {
    case T_CONST:
        switch ($this->getVisibilityForToken($file, $pointer)) {
            case T_PUBLIC:
                return self::STAGE_PUBLIC_CONSTANTS;
            case T_PROTECTED:
                return self::STAGE_PROTECTED_CONSTANTS;
            default:
                return self::STAGE_PRIVATE_CONSTANTS;
        }
}
```

As you can see, there's no way the code would continue executing after running inner `switch`. However, this error is reported 

> There must be a comment when fall-through is intentional in a non-empty case body

on line `    case T_CONST:`.

Changing the code this way satisfies the sniff:

```diff
            case T_CONST:
                switch ($this->getVisibilityForToken($file, $pointer)) {
                    case T_PUBLIC:
                        return self::STAGE_PUBLIC_CONSTANTS;
                    case T_PROTECTED:
                        return self::STAGE_PROTECTED_CONSTANTS;
                    default:
                        return self::STAGE_PRIVATE_CONSTANTS;
                }
+
+                // no way to get here
```

**Comment by gsherwood** [comment]
The sniff doesn't attempt to figure out if every possible execution path would lead to an exit condition, which is why you get this error. The intention of the sniff is to always provide the comment so that it is clear what is happening in the code without a developer also having to follow all the code paths.

I'm going to close this as it's not a use case that this sniff has been designed for and the effort to attempt to support it is too high for me to take on.

**Comment by gsherwood** [comment]
PR #3186 has now implemented this request. It will be released in version 3.6.0.
